### PR TITLE
Added Testimonial component

### PR DESCRIPTION
**Pull request by ZacheryEC001**
This pull request aims to resolve issue #56 by implementing the testimonial component and it's options. Everything from that issue is addressed by this pull request, with the exception of the starGradient option. Tailwind does not seem to support directly filling SVGs with gradients, so for the sake of time I will address this concern in a future pull request. I have opened up issue #79 as a result.

## Files Added

**Testimonial.jsx** 
* Implemented the Testimonial component

**TestimonialOptions.jsx**
* Implemented the TestimonialOptions

**TestimonialPreviewLight.png**
* Preview for the Testimonial component which shows in the ComponentListTab

**star.svg**
* SVG file for the stars in the testimonial

## Files Modified

**registeredComponents.js** 
* Added testimonial component to the registered components list

## Screenshots

**Light Mode**
image

**Dark Mode**
image

### CODE REVIEW

**Review by Lettuce05 — CHANGES_REQUESTED** [review]
Everything looks great with the code and it functionally works when changing the options in the UI. The only fix that I would recommend is that the component itself is not responsive. When I make the screen skinnier the component should adapt so that the content fits on the page rather than being pushed off the screen. My suggestion would be to set the width of the component to 80% or 90% for small screen sizes and then keep your current styling for larger screen sizes.

**Comment by Lettuce05** [comment]
[Open in Codeflow](https://pr.new/github.com/cs472-team6/cs472-team6/pull/80)

**Review by ebacolod — CHANGES_REQUESTED** [review]
The testimonial component takes up the whole width of the rendering page when at 2xl screen sizes. 
image

This link explains how tailwind does its responsive design:
https://tailwindcss.com/docs/responsive-design#working-mobile-first

Also, I suggest looking at the Service Wrapper component to see how the sizing is done. (Specifically in the ServiceWrapper.jsx file).

Once you're ready, go ahead and make changes to Testimonial.jsx.

**Comment by ZacheryEC001** [comment]
> The testimonial component takes up the whole width of the rendering page when at 2xl screen sizes. image
> 
> This link explains how tailwind does its responsive design: https://tailwindcss.com/docs/responsive-design#working-mobile-first
> 
> Also, I suggest looking at the Service Wrapper component to see how the sizing is done. (Specifically in the ServiceWrapper.jsx file).
> 
> Once you're ready, go ahead and make changes to Testimonial.jsx.

Fixed

**Review by ebacolod — APPROVED** [review]
2xl bug is fixed. Good Work!

File: src/utils/registeredComponents.js [hunk 1]
```diff
@@ -15,8 +19,9 @@ import CtaSectionPreview from "../components/CtaSection/CtaSectionPreview.png";
 import { StatsWrapperComponent } from "../components/StatsWrapper/StatsWrapperComponent";
 import { contactSectionComponent } from "../components/ContactSection/contactSectionComponent";
 
-
 export const registeredComponents = new Map([
+    ['Banner', new Component('Banner', Banner, BannerOptions, BannerPreview, {words: 'Design,Develop,Discover', gradient: Gradients.default})],
+    ['Testimonial', TestimonialComponent]
```
Comment (Lettuce05) [inline src/utils/registeredComponents.js:24]: Missing comma which is causing a syntax error for the application.

**Review by Lettuce05 — CHANGES_REQUESTED** [review]

**Review by Lettuce05 — APPROVED** [review]
The component now works great at all sizes. As can be seen by the images below for each of the screen sizes. This pull request is good to be merged. No functionality has broken as I can still dynamically update the options of the component (this can be seen in the images as well).

### Large Screen (Desktop)
image

### Medium Screen (Tablet)

image

### Small Screen (Phone)

image

**Pull request merged**